osu.Game/Screens/Menu/MainMenu.cs

```diff
@@ -314,13 +305,28 @@ private bool onLogoClick(Func<bool> originalAction)
                     dialogOverlay.Push(new MobileDisclaimerDialog(() =>
                     {
                         showMobileDisclaimer.Value = false;
+                        displayLoginIfApplicable();
```

Comment:
This surprised me at first but this will also work if the dialog is dismissed by clicking anywhere outside the dialog, because there's internal logic in `PopupDialog` that clicks the last button in the dialog, which the singular "Understood" button in this case.